### ISSUE TITLE
"No content to map due to end-of-input"

### ISSUE THREAD

**Issue by javagc** [jesperfj/force-rest-api#32]
Hi, thanks for this library, i am trying to connect force API, my code is here

```
 ForceApi api = new ForceApi(new ApiConfig()
                .setUsername("username")
                .setPassword("password")
                .setClientId("client_id")
                .setClientSecret("client_secret"));
```

result is 

```
Exception in thread "main" java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: No content to map due to end-of-input
 at [Source: UNKNOWN; line: 1, column: 1]
```

**Comment by javagc** [comment]
So problem was Ip Restriction, 
1. Click Setup in the top-right
2. Select Administer > Security Controls > Network Access from the left navigation
3. Click New
4. Add your ip address range
5. Click Save

**Comment by pcornish** [comment]
This happens whenever authentication fails; it would be good if this exception could be trapped specifically by the library and an particular exception thrown so it can be handled.

**Comment by jesperfj** [comment]
Thanks for the feedback. Could you take a look at #34 and see if you think that PR will improve things for future users in your situation?

**Comment by jesperfj** [comment]
Closing with changes released in (0.0.32)[https://github.com/jesperfj/force-rest-api/releases/tag/force-rest-api-0.0.32]
